### PR TITLE
feat: support glob pattern in operation filepath

### LINKED ISSUE #794: [Feature] : Use Directory Path Ref in operation like apply, assert etc.
### Problem Statement

Currently we support the struct : `FileRefOrResource` in which we can only specify the file reference not the complete directory. 
Which makes us to write all the files explicitly which we want to use in a particular operation. 

### Solution Description

Use something like `directoryRef` in which we can specify the path to the directory in which we contain all the manifest to be applied during the operation.

### Alternatives

_No response_

### Additional Context

_No response_

### Slack discussion

https://kubernetes.slack.com/archives/C067LUFL43U/p1706995405247079

### Research

- [X] I have searched other issues in this repository and mine is not recorded.

### PR DESCRIPTION
**Pull request by shubham-cmyk**
## Related issue
Fixes #794

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/kyverno/kyverno/blob/main/CONTRIBUTING.md).
- [x] I have read the [PR documentation guide](https://github.com/kyverno/kyverno/blob/main/.github/pr_documentation.md) and followed the process including adding proof manifests to this PR.
- [ ] This is a bug fix and I have added unit tests that prove my fix is effective.

### CODE REVIEW

File: pkg/resource/load.go [hunk 1]
```diff
@@ -22,18 +22,36 @@ type (
...
+		tests, err := Parse(content, manifest)
+		if err != nil {
+			return err
+		}
+		if len(tests) == 0 {
+			return fmt.Errorf("found no resource in %s", path)
+		}
+		resources = append(resources, tests...)
+		return nil
+	})
```
Comment (eddycharly) [inline pkg/resource/load.go:47]: Would be easier to first lookup files and then read/parse them.

File: pkg/resource/load.go [hunk 2]
```diff
@@ -22,18 +22,32 @@ type (
 )
 
 func Load(path string, manifest bool) ([]unstructured.Unstructured, error) {
-	content, err := os.ReadFile(filepath.Clean(path))
+	var resources []unstructured.Unstructured
+
```
Comment (eddycharly) [inline pkg/resource/load.go:26]: Please, remove empty line =)

**Comment by eddycharly** [comment]
Please update release notes ;)

**Comment by shubham-cmyk** [comment]
Added Release Notes

**Comment by eddycharly** [comment]
@shubham-cmyk add release notes in `main.md`

**Comment by eddycharly** [comment]
@shubham-cmyk can you fix conflict ?

File: .release-notes/main.md [hunk 3]
```diff
@@ -24,11 +25,9 @@ Release notes for `TODO`.
 
 ## 📚 Docs 📚
 
-- Added community meetings docs
-- Added google groups in community docs and README
-- Removed all references to `TestStep` in the docs as this is not supported anymore
+- Remove TestStep reference examples
```
Comment (eddycharly) [inline .release-notes/main.md:28]: Don't remove existing entries please

Comment (shubham-cmyk) [reply .release-notes/main.md:28]: it was removed by mistake

Comment (eddycharly) [reply .release-notes/main.md:28]: Ok add back and we can merge

File: .release-notes/main.md [hunk 4]
```diff
@@ -27,8 +28,11 @@ Release notes for `TODO`.
 - Added community meetings docs
 - Added google groups in community docs and README
 - Removed all references to `TestStep` in the docs as this is not supported anymore
+- Remove TestStep reference examples
 
 ## 🎸 Misc 🎸
 
 - Added kubebuilder community recording link where chainsaw was presented
 - Added a `MAINTAINERS.md` file
+- Added makefile target to check go code coverage
```
Comment (eddycharly) [inline .release-notes/main.md:37]: Remove this line, it's not relevant imho

Comment (shubham-cmyk) [reply .release-notes/main.md:37]: done

**Review by eddycharly — APPROVED** [review]

**Pull request merged**